runtime/compiler/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -6505,7 +6505,7 @@ TR::Register *J9::Power::TreeEvaluator::VMnewEvaluator(TR::Node *node, TR::CodeG
          }
 
       //Do not need zero init if using dualTLH now.
-      if (!isDualTLH && needZeroInit && (!isConstantZeroLenArrayAlloc))
+      if (needZeroInit && (!isConstantZeroLenArrayAlloc))
          {
```

Comment:
I am not sure why initially isDualTLH was considered as it is.  it appeared to me that the extra checking was just added pessimistically somehow. i will ask you to think through it whether your fix is correct in all cases (dualTLH enabled or disabled, and batchClearing enabled or disabled).